server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java

```diff
@@ -341,58 +341,25 @@ public void parse(
-                            indexRequestConsumer.accept(
-                                new IndexRequest(index).id(id)
-                                    .routing(routing)
-                                    .version(version)
-                                    .versionType(versionType)
-                                    .setPipeline(pipeline)
-                                    .setIfSeqNo(ifSeqNo)
-                                    .setIfPrimaryTerm(ifPrimaryTerm)
-                                    .source(sliceTrimmingCarriageReturn(data, from, nextMarker, xContentType), xContentType)
-                                    .setDynamicTemplates(dynamicTemplates)
-                                    .setRequireAlias(requireAlias)
-                                    .setRequireDataStream(requireDataStream)
-                                    .setListExecutedPipelines(listExecutedPipelines),
-                                type
-                            );
-                        } else {
-                            indexRequestConsumer.accept(
-                                new IndexRequest(index).id(id)
-                                    .routing(routing)
-                                    .version(version)
-                                    .versionType(versionType)
-                                    .create("create".equals(opType))
-                                    .setPipeline(pipeline)
-                                    .setIfSeqNo(ifSeqNo)
-                                    .setIfPrimaryTerm(ifPrimaryTerm)
-                                    .source(sliceTrimmingCarriageReturn(data, from, nextMarker, xContentType), xContentType)
-                                    .setDynamicTemplates(dynamicTemplates)
-                                    .setRequireAlias(requireAlias)
-                                    .setRequireDataStream(requireDataStream)
-                                    .setListExecutedPipelines(listExecutedPipelines),
-                                type
-                            );
+                    if ("index".equals(action) || "create".equals(action)) {
+                        var indexRequest = new IndexRequest(index).id(id)
+                            .routing(routing)
+                            .version(version)
+                            .versionType(versionType)
+                            .setPipeline(pipeline)
+                            .setIfSeqNo(ifSeqNo)
+                            .setIfPrimaryTerm(ifPrimaryTerm)
+                            .source(sliceTrimmingCarriageReturn(data, from, nextMarker, xContentType), xContentType)
+                            .setDynamicTemplates(dynamicTemplates)
+                            .setRequireAlias(requireAlias)
+                            .setRequireDataStream(requireDataStream)
+                            .setListExecutedPipelines(listExecutedPipelines);
+                        if ("create".equals(action)) {
```

Comment:
```suggestion
                        if (opType != null && "create".equals(opType)) {
```

Reply:
Not quite right? The old logic would set `true` if the action is "create" or if the op-type is create? Doing a null check before putting something on the right hand side of the equals seems unnecessary too?

Reply:
What I wanted to bring to your attention is that the current PR misses the path of the old code:
`"index".equals(action) -> opType != null -> .create("create".equals(opType))`. Right?

Reply:
Ah got it, that's actually covered 3 lines down in https://github.com/elastic/elasticsearch/pull/109302/files#diff-07b6912d7b05ace64d467c90faf865f31d682835a3b2c4d044c6a8db621f7481R360 

In this block we are either "index" or "create" so the `else if` is only taken for "index". That should replicate the old behavior 1:1 I think?

Reply:
Ah yes! Thanks for pointing out to me explicitly, for some reason I did not see it immediately.